packages/debug-helper/modules/inc/mockers/class-nonces-runner.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * The Mocker Runner that creates mock nonces.
+ *
+ * @package Jetpack
+ */
+
+namespace Automattic\Jetpack\Debug_Helper\Mocker;
+
+require_once __DIR__ . '/class-tools.php';
+
+/**
+ * Creating the mock nonces.
+ */
+class Nonces_Runner implements Runner_Interface {
+
+	/**
+	 * Generate random nonces.
+	 *
+	 * @param int $number Number of nonces to generate.
+	 *
+	 * @return bool
+	 */
+	public function run( $number ) {
+		for ( $i = $number, $per_batch = 500; $i > 0; $i -= $per_batch ) {
+			$this->run_batch( min( $per_batch, $i ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Add a batch of mock options.
+	 *
+	 * @param int $limit How many options to add.
+	 *
+	 * @return bool
+	 */
+	private function run_batch( $limit ) {
+		global $wpdb;
+
+		if ( ! $limit ) {
+			return false;
+		}
+
+		$sql = "INSERT INTO {$wpdb->prefix}options (option_name, option_value, autoload) VALUES ";
+
+		for ( $i = 0; $i < $limit; ++$i ) {
+			list( $name, $value ) = $this->get_random_nonce();
+			$sql                 .= "('" . esc_sql( $name ) . "', '" . esc_sql( $value ) . "', 'no'), ";
+		}
+
+		// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
```

Comment:
This can be moved to a prepared statement. It might look something like this: (Disclaimer: haven't tested it in a real env, but should give a good idea)

```php

$sql = "INSERT INTO {$wpdb->prefix}options (option_name, option_value, autoload) VALUES ";
$sql .= implode( ', ', array_fill( 0, $limit, '( %s, %s, "no" )' ) );

$values_to_insert = array();
for ( $i = 0; $i < $limit; ++$i ) {
    $values_to_insert = array_merge( $values_to_insert, $this->get_random_nonce() );
}

$prepared_query = $wpdb->prepare( $sql, $values_to_insert );

$wpdb->query( $prepared_query );

```

Reply:
Thanks @bisko.
I was concerned about the performance, and after measuring the timing it turned out that `$wpdb->prepare()` is quite fast, so the difference is negligible.

Fixed now.

Reply:
Awesome! Can we remove the `phpcs:ignore` statement too? :)

Reply:
Unfortunately, the `NotPrepared` error is still thrown, so we need it.
Apparently because the SQL code is passed via a variable.

We probably need to address that linting problem later on.